src/bind.py

```diff
@@ -199,17 +253,25 @@ def handle_relation_data(
         # Create staging area
         tempdir = tempfile.mkdtemp()
 
-        # Write zone files
-        zones = {}
+        # Create zones list
+        zones: typing.List[Zone] = []
         for record_requirer_data, _ in relation_data:
-            zones.update(self._to_bind_zones(record_requirer_data))
-        logger.debug("ZONES: %s", zones)
-        for name, content in zones.items():
+            zones.extend(self._record_requirer_data_to_zones(record_requirer_data))
+
+        # Check for conflicts
+        _, conflicting = self._get_conflicts(zones)
+        if len(conflicting) > 0:
+            return self._create_dns_record_provider_data(relation_data)
+
+        # Write zone files
+        zone_files: typing.Dict[str, str] = self._zones_to_files(zones)
+        for name, content in zone_files.items():
```

Comment:
I'm wondering if we could do 
```suggestion
    with tempfile.TemporaryDirectory() as tmpdirname:
        ...
```
:)

Reply:
(this is also a nitpick, please feel free to close)

Reply:
I suppose that you meant to replace `tempdir = tempfile.mkdtemp()`. If so, I agree

Reply:
Fixed